lib/bson/bson.js

```diff
@@ -49,6 +52,14 @@ BSON.prototype.serialize = function serialize(object, options) {
     typeof options.serializeFunctions === 'boolean' ? options.serializeFunctions : false;
   var ignoreUndefined =
     typeof options.ignoreUndefined === 'boolean' ? options.ignoreUndefined : true;
+  var minInternalBufferSize =
+    typeof options.minInternalBufferSize === 'number' ? options.minInternalBufferSize : MAXSIZE;
+  
+  // Resize the internal serialization buffer if needed
+  if (currentBufferSize < minInternalBufferSize) {
```

Comment:
do we need to keep an extra variable around for this? Doesn't `if (buffer.length < minInternalBufferSize)` suffice?

Reply:
good point